### ISSUE TITLE
[Feature][flink-connector-jdbc] JdbcSink query sql supports param placeholder replacing row index.  

### ISSUE THREAD

**Issue by kalencaya** [apache/seatunnel#1829]
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

a typical `JdbcSink` config follows:
```
JdbcSink {
    source_table_name = fake
    driver = com.mysql.jdbc.Driver
    url = "jdbc:mysql://localhost/test"
    username = root
    query = "insert into test(name,age) values(?,?)"
    batch_size = 2
}
```
`insert into test(name,age) values(?,?)` requires that `Row` fields must match `name, age` order as `JdbcUtils#setRecordToStatement` implemention:
```
    public DataStreamSink<Row> outputStream(FlinkEnvironment env, DataStream<Row> dataStream) {
        Table table = env.getStreamTableEnvironment().fromDataStream(dataStream);
        TypeInformation<?>[] fieldTypes = table.getSchema().getFieldTypes();
        int[] types = Arrays.stream(fieldTypes).mapToInt(JdbcTypeUtil::typeInformationToSqlType).toArray();
        SinkFunction<Row> sink = org.apache.flink.connector.jdbc.JdbcSink.sink(
            query,
            (st, row) -> JdbcUtils.setRecordToStatement(st, types, row), // the key reason
            JdbcExecutionOptions.builder()
                .withBatchSize(batchSize)
                .withBatchIntervalMs(batchIntervalMs)
                .withMaxRetries(maxRetries)
                .build(),
            new JdbcConnectionOptions.JdbcConnectionOptionsBuilder()
                .withUrl(dbUrl)
                .withDriverName(driverName)
                .withUsername(username)
                .withPassword(password)
                .build());

        if (config.hasPath(PARALLELISM)) {
            return dataStream.addSink(sink).setParallelism(config.getInt(PARALLELISM));
        }
        return dataStream.addSink(sink);
    }
```
now query should support param placeholder such as `insert into test(name,age) values(#{name},#{age})`  would be better for  two reasons:
* more brief semantics
* jdbc upsert support. `insert into xxx on duplicate xxx` and `replace into` grammar will benefit

### Usage Scenario

flink-connector-jdbc supports upsert

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by ruanwenjun** [comment]
>1: `insert into test(name,age) values(#{name},#{age}) ` is more brief semantics.

I don't think so, I think `insert into test(name,age) values(?,?)` is more easy and is standard SQL.
>2: jdbc upsert support

This is interesting, but I would want to know if this works, I guess it will only be able to upsert in the current jdbc batch?

**Comment by kalencaya** [comment]
there are diferent opinions on `brief semantics`, where I matter more on `semantics` for reading and understanding and of course `(?, ?, ?)` is clear and brief.
`insert into test(field1, field2, field3....) values (?,?,?...)` works with `PreparedStatement` not users, `insert into test(field1, field2, field3....) values (#{field1}, #{field2}, #{field2}, ...)` is more direct for user define how fields update.

**Comment by kalencaya** [comment]
I'm confused with the meaning of `the current jdbc batch`.
actually, I have added a transform operator on datastream or dataset which mapping Row(field1, filed2) to Row(field1, field2, field1, field2) dependending on `#{}` placeholder, I think it can work always.
the code before:
```
dataSet.output(format);
```
the code after:
```
dataSet.map(source -> copyFields(source, fieldPositionMap)).returns(rowTypeInfo).output(format);
```

**Comment by Hisoka-X** [comment]
I agree with @kalencaya . The `(?, ?, ?)` not clear in seatunnel config file when create an data flow. Make a `#{}` placeholder will be good choose. But the `(?, ?, ?)` should work too.

**Comment by kalencaya** [comment]
> I agree with @kalencaya . The `(?, ?, ?)` not clear in seatunnel config file when create an data flow. Make a `#{}` placeholder will be good choose. But the `(?, ?, ?)` should work too.

emmmm, try to keep compatible with the existing `(?, ?, ?)` style? Does it is necessary?
